src/hooks/useInactiveListener.ts

```diff
@@ -26,11 +37,13 @@ function useInactiveListener(suppress = false) {
                     activate(injected, undefined, true).catch(error => {
                         console.error('Failed to activate after accounts changed', error)
                     })
+                } else {
+                  deactivate()
                 }
             }
 
-            ethereum.on('chainChanged', handleChainChanged)
-            ethereum.on('accountsChanged', handleAccountsChanged)
+            ethereum.selectedProvider.on('chainChanged', handleChainChanged)
+            ethereum.selectedProvider.on('accountsChanged', handleAccountsChanged)
 
             return () => {
                 if (ethereum.removeListener) {
```

Comment:
selectedPRovider?

Reply:
Ontopic: see comment index.ts

Off-topic:
O, about useInactiveListener
I believe this file could be deleted, as the logic has been moved to index and is not referenced anywhere
I just changed it here as well, just to be sure in case I missed anything